### PR TITLE
Expand docs for `ASYNC109`

### PR DESCRIPTION
**Pull request by jamesbraza**
## Summary

Documenting when to use or not use ASYNC109 a bit more.

Closes https://github.com/astral-sh/ruff/issues/12353

## Test Plan

It wasn't tested

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs [hunk 1]
```diff
@@ -11,10 +11,18 @@ use crate::settings::types::PythonVersion;
...
 /// Rather than implementing asynchronous timeout behavior manually, prefer
-/// built-in timeout functionality, such as `asyncio.timeout`, `trio.fail_after`,
-/// or `anyio.move_on_after`, among others.
+/// built-in timeout functionality, such as `asyncio.timeout`,
+/// `trio.fail_after`, or `anyio.move_on_after`, among others.
+///
+/// If you are abstracting `asyncio` within a function
+/// that internally utilizes `asyncio.timeout`,
+/// this rule can be disabled, or use a more specific timeout name
+/// such as `your_behavior_timeout`.
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs:25]: "If the rule has false positives, you can disable it" feels like quite general advice that could be applied to any rule ;) I'm not sure we need to say that; it's good to keep the docs concise where possible.

Maybe something like this?

```suggestion
/// For functions that wrap `asyncio.timeout`,
/// false positives from this rule can be avoided by using
/// a different parameter name.
```

**Review by AlexWaygood — COMMENTED** [review]
Thanks! I agree the docs for this function can be improved, and I think this is a step in the right direction overall. However, I'm not sure the PR fully addresses @jakkdl's comments in https://github.com/astral-sh/ruff/issues/12353#issuecomment-2245666064 and https://github.com/astral-sh/ruff/issues/12353#issuecomment-2296575231. It would be great if the rule's docs could attempt to give a proper explanation of why the rule exists in the first place (to help enforce opinionated design decisions relating to a broader philosophy of how structured concurrency works), probably linking to https://vorpus.org/blog/some-thoughts-on-asynchronous-api-design-in-a-post-asyncawait-world/#timeouts-and-cancellation.

I guess those updates don't necessarily need to be made in this PR, but I don't think this PR is sufficient to close out #12353 as it currently stands

File: crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs [hunk 2]
```diff
@@ -11,10 +11,18 @@ use crate::settings::types::PythonVersion;
 /// ## What it does
 /// Checks for `async` functions with a `timeout` argument.
 ///
+/// This rule is only applied for Python 3.11 and newer,
+/// as `asyncio.timeout` was added in Python 3.11.
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs]: Oh nice catch here, yeah I see what you're saying. I just updated the wording for that

File: crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs (hunk 1 above)
Comment (jamesbraza) [reply crates/ruff_linter/src/rules/flake8_async/rules/async_function_with_timeout.rs]: Thanks for the great suggestion, implemented

**Comment by jamesbraza** [comment]
Alright, I just added a small blurb mentioning structured concurrency and linking that blog post 👌 

Thanks for the detailed and thoughtful review

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**